File: array/find-the-highest-altitude.js
var largestAltitude = function (gain) {
  let sum = 0,
    max = 0;
  for (let i = 0; i < gain.length; i++) {
    sum += gain[i];
    max = Math.max(max, sum);
  }

  return max;
};

console.log(largestAltitude([-5,1,5,0,-7]));

// console.log(Math.max(0, ...[-4, -3, -2, -1, 4, 3, 2]));

File: Java_leetcode/findTheHighestAltitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int max_alt=0;
        int curr_alt=0;
        for(int i=0;i<gain.length;i++){
            curr_alt+=gain[i]; 
            max_alt=Math.max(curr_alt, max_alt);
        }
        return max_alt; 
    }

    public static void main(String[] args){
        int[] arr =new int[]{-4,-3,-2,-1,4,3,2};
        Solution s1 = new Solution();
        System.out.println(s1.largestAltitude(arr));
    }
}